frappe/twofactor.py

```diff
@@ -463,7 +463,8 @@ def disable():
 def reset_otp_secret(user):
 	otp_issuer = frappe.db.get_value("System Settings", "System Settings", "otp_issuer_name")
 	user_email = frappe.db.get_value("User", user, "email")
-	if frappe.session.user in ["Administrator", user]:
+	user_roles = frappe.get_roles()
+	if "System Manager" in user_roles or frappe.session.user == user:
```

Comment:
Simpler condition should be first:

```suggestion
	if frappe.session.user == user or "System Manager" in frappe.get_roles():
```

Reply:
Hi @sagarvora thanks for your feedback, I updated the PR accordingly.